### PR TITLE
Build jar package before building deb and/or rpm

### LINKED ISSUE #890: Building with dockerbuild_build does not build JAR
[Project board link](https://github.com/orgs/k8ssandra/projects/8/views/1?pane=issue&itemId=13937794)

### How to reproduce
Build without using Maven cache. Either delete your Maven cache or remove the .m2 volume in docker-compose.yaml.

Follow these instructions: http://cassandra-reaper.io/docs/download/docker/

```
cd src/packaging/docker-build
docker-compose build
docker-compose run build
```

You should get the following output:

```
$ docker-compose run build
+ cd /usr/src/app/cassandra-reaper
++ printf 'VER\t${project.version}'
++ mvn help:evaluate
++ grep '^VER'
++ cut -f2
+ export VERSION=2.1.0-SNAPSHOT
+ VERSION=2.1.0-SNAPSHOT
+ cd /usr/src/app/cassandra-reaper/src/packaging
+ make build-packages
echo "2.1.0-SNAPSHOT..."
2.1.0-SNAPSHOT...
rm -Rf packages/*
rm -Rf build/*
mkdir -p build/usr/share/cassandra-reaper
mkdir -p build/usr/local/bin
mkdir -p build/etc/init.d
mkdir -p build/etc/cassandra-reaper
mkdir -p build/etc/cassandra-reaper/configs
mkdir -p build/etc/bash_completion.d
mkdir -p build/lib/systemd/system/
cp resource/cassandra-reaper.yaml build/etc/cassandra-reaper/
cp resource/cassandra-reaper*.yaml build/etc/cassandra-reaper/configs
cp resource/cassandra-reaper-ssl.properties build/etc/cassandra-reaper/configs
cp ../server/target/cassandra-reaper-2.1.0-SNAPSHOT.jar build/usr/share/cassandra-reaper/
cp: cannot stat '../server/target/cassandra-reaper-2.1.0-SNAPSHOT.jar': No such file or directory
Makefile:20: recipe for target 'prepare' failed
make: *** [prepare] Error 1
```

### Solution
The issue is that the entrypoint only executes the "build-packages" goal, and this goal does not build the JAR.

Since the deb and rpm packages requires JAR package, I suggest the "deb" and "rpm" goals should depend on "package" goal.

┆Issue is synchronized with this [Jira Story](https://datastax.jira.com/browse/REAP-135) by [Unito](https://www.unito.io)
┆Issue Number: REAP-135

### PR DESCRIPTION
**Pull request by simonfontana**
Fixes #890

### CODE REVIEW

File: src/packaging/Makefile [hunk 1]
```diff
@@ -16,7 +16,7 @@
 package:
```
Comment (ossarga) [inline src/packaging/Makefile:16]: What if we also include the JAR path as a target?

```suggestion
REAPER_JAR=../server/target/cassandra-reaper-$(VERSION).jar

$(REAPER_JAR) package:
```

File: src/packaging/Makefile [hunk 2]
```diff
@@ -16,7 +16,7 @@
...
 	mvn -B clean package  -f ../../pom.xml -DskipTests
 
-prepare:
+prepare: package
```
Comment (ossarga) [inline src/packaging/Makefile:19]: Perhaps this target should depend on the JAR path? That way if the JAR exists, we just do the `prepare` steps. If the JAR file is missing we then go and build it.

```suggestion
prepare: $(REAPER_JAR)
```

We should also change the `all` target to depend on the JAR path in a similar way as well.

**Review by ossarga — CHANGES_REQUESTED** [review]
@simonfontana thank you for submitting this patch!

A while back commit [c0a980c](https://github.com/thelastpickle/cassandra-reaper/commit/c0a980c9d0c643199c00db9de1bff939a864a944)  was made to avoid unnecessary building of the JAR files.

I agree in principle with the proposed patch. I think it needs to do more to take into account the previous issue with running the container. Specifically, as per my comments below the `prepare` target should depend on the JAR file path rather than the `package` target. In addition, the JAR file path should be a target that performs the same actions as `package`.

**Comment by simonfontana** [comment]
@ossarga thanks for the review. I tried to apply your changes and verify it but after the merge with master I have some issues. They are unrelated to the change and I'm not invested enough in the project currently that I have time to sort them out. Perhaps you can help. 

```
$ docker-compose run build
Creating docker-build_build_run ... done
.........
(omitted)
.........
Hash: 64092869d3564e14cfa8
Version: webpack 4.46.0
Time: 10209ms
Built at: 03/26/2022 7:20:05 PM
 8 assets
Entrypoint index = index.js
Entrypoint schedules = schedules.js
Entrypoint repair = repair.js
Entrypoint snapshot = snapshot.js
Entrypoint segments = segments.js
Entrypoint login = login.js
Entrypoint events = events.js
Entrypoint deps = deps.js
[527] multi ./app/index.js 28 bytes {2} [built]
[528] ./app/index.js 3.68 KiB {2} [built]
[533] multi ./app/schedules.js 28 bytes {5} [built]
[534] ./app/schedules.js 3.49 KiB {5} [built]
[548] multi ./app/repair.js 28 bytes {4} [built]
[549] ./app/repair.js 3.63 KiB {4} [built]
[552] multi ./app/snapshot.js 28 bytes {7} [built]
[553] ./app/snapshot.js 3.16 KiB {7} [built]
[556] multi ./app/segments.js 28 bytes {6} [built]
[557] ./app/segments.js 2.81 KiB {6} [built]
[558] multi ./app/login.js 28 bytes {3} [built]
[559] ./app/login.js 2.52 KiB {3} [built]
[560] multi ./app/events.js 28 bytes {1} [built]
[561] ./app/events.js 3.81 KiB {1} [built]
[567] multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css 196 bytes {0} [built]
    + 577 hidden modules

ERROR in multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css
Module not found: Error: Can't resolve 'bootstrap-multiselect.css' in '/usr/src/app/cassandra-reaper/src/ui'
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css deps[12]

ERROR in ./app/jsx/event-list.jsx
Module not found: Error: Can't resolve 'datatables.net' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/event-list.jsx 27:19-44
 @ ./app/jsx/event-screen.jsx
 @ ./app/events.js
 @ multi ./app/events.js

ERROR in multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css
Module not found: Error: Can't resolve 'datatables.net-bs.css' in '/usr/src/app/cassandra-reaper/src/ui'
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css deps[13]

ERROR in ./app/jsx/event-list.jsx
Module not found: Error: Can't resolve 'datatables.net-bs.js' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/event-list.jsx 31:23-54
 @ ./app/jsx/event-screen.jsx
 @ ./app/events.js
 @ multi ./app/events.js

ERROR in multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css
Module not found: Error: Can't resolve 'font-awesome.css' in '/usr/src/app/cassandra-reaper/src/ui'
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css deps[3]

ERROR in multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui'
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css deps[6]

ERROR in ./app/segments.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app'
 @ ./app/segments.js 3:14-31
 @ multi ./app/segments.js

ERROR in ./app/login.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app'
 @ ./app/login.js 3:14-31
 @ multi ./app/login.js

ERROR in ./app/schedules.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app'
 @ ./app/schedules.js 3:14-31
 @ multi ./app/schedules.js

ERROR in ./app/repair.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app'
 @ ./app/repair.js 3:14-31
 @ multi ./app/repair.js

ERROR in ./app/events.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app'
 @ ./app/events.js 3:14-31
 @ multi ./app/events.js

ERROR in ./app/observable.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app'
 @ ./app/observable.js 12:14-31
 @ ./app/index.js
 @ multi ./app/index.js

ERROR in ./app/index.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app'
 @ ./app/index.js 3:14-31
 @ multi ./app/index.js

ERROR in ./app/snapshot.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app'
 @ ./app/snapshot.js 3:14-31
 @ multi ./app/snapshot.js

ERROR in ./app/jsx/cluster-list.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/cluster-list.jsx 37:14-31
 @ ./app/jsx/cluster-screen.jsx
 @ ./app/index.js
 @ multi ./app/index.js

ERROR in ./app/jsx/dropped-messages.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/dropped-messages.jsx 25:14-31
 @ ./app/jsx/node-status.jsx
 @ ./app/jsx/cluster-list.jsx
 @ ./app/jsx/cluster-screen.jsx
 @ ./app/index.js
 @ multi ./app/index.js

ERROR in ./app/jsx/snapshot.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/snapshot.jsx 29:14-31
 @ ./app/jsx/node-status.jsx
 @ ./app/jsx/cluster-list.jsx
 @ ./app/jsx/cluster-screen.jsx
 @ ./app/index.js
 @ multi ./app/index.js

ERROR in ./app/jsx/event-screen.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/event-screen.jsx 1:0-17
 @ ./app/events.js
 @ multi ./app/events.js

ERROR in ./app/jsx/snapshot-screen.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/snapshot-screen.jsx 1:0-17
 @ ./app/snapshot.js
 @ multi ./app/snapshot.js

ERROR in ./app/jsx/segment-list.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/segment-list.jsx 33:14-31
 @ ./app/segments.js
 @ multi ./app/segments.js

ERROR in ./app/jsx/client-request-latency.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/client-request-latency.jsx 33:14-31
 @ ./app/jsx/node-status.jsx
 @ ./app/jsx/cluster-list.jsx
 @ ./app/jsx/cluster-screen.jsx
 @ ./app/index.js
 @ multi ./app/index.js

ERROR in ./app/jsx/cluster-selection.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/cluster-selection.jsx 1:0-17
 @ ./app/jsx/event-screen.jsx
 @ ./app/events.js
 @ multi ./app/events.js

ERROR in ./app/jsx/event-list.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/event-list.jsx 19:14-31
 @ ./app/jsx/event-screen.jsx
 @ ./app/events.js
 @ multi ./app/events.js

ERROR in ./app/jsx/streams.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/streams.jsx 29:14-31
 @ ./app/jsx/node-status.jsx
 @ ./app/jsx/cluster-list.jsx
 @ ./app/jsx/cluster-screen.jsx
 @ ./app/index.js
 @ multi ./app/index.js

ERROR in ./app/jsx/event-subscription-form.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/event-subscription-form.jsx 7:14-31
 @ ./app/jsx/event-screen.jsx
 @ ./app/events.js
 @ multi ./app/events.js

ERROR in ./app/jsx/snapshot-list.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/snapshot-list.jsx 1:0-17
 @ ./app/jsx/snapshot-screen.jsx
 @ ./app/snapshot.js
 @ multi ./app/snapshot.js

ERROR in ./app/jsx/snapshot-form.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/snapshot-form.jsx 25:14-31
 @ ./app/jsx/snapshot-screen.jsx
 @ ./app/snapshot.js
 @ multi ./app/snapshot.js

ERROR in ./app/jsx/schedule-list.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/schedule-list.jsx 1:0-17
 @ ./app/jsx/schedule-screen.jsx
 @ ./app/schedules.js
 @ multi ./app/schedules.js

ERROR in ./app/jsx/repair-form.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/repair-form.jsx 25:14-31
 @ ./app/jsx/schedule-screen.jsx
 @ ./app/schedules.js
 @ multi ./app/schedules.js

ERROR in ./app/jsx/node-status.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/node-status.jsx 77:14-31
 @ ./app/jsx/cluster-list.jsx
 @ ./app/jsx/cluster-screen.jsx
 @ ./app/index.js
 @ multi ./app/index.js

ERROR in ./app/jsx/tpstats.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/tpstats.jsx 25:14-31
 @ ./app/jsx/node-status.jsx
 @ ./app/jsx/cluster-list.jsx
 @ ./app/jsx/cluster-screen.jsx
 @ ./app/index.js
 @ multi ./app/index.js

ERROR in ./app/jsx/active-compactions.jsx
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/app/jsx'
 @ ./app/jsx/active-compactions.jsx 29:14-31
 @ ./app/jsx/node-status.jsx
 @ ./app/jsx/cluster-list.jsx
 @ ./app/jsx/cluster-screen.jsx
 @ ./app/index.js
 @ multi ./app/index.js

ERROR in ./node_modules/bootstrap/js/alert.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/node_modules/bootstrap/js'
 @ ./node_modules/bootstrap/js/alert.js 1:0-17
 @ ./node_modules/bootstrap/dist/js/npm.js
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css

ERROR in ./node_modules/bootstrap/js/modal.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/node_modules/bootstrap/js'
 @ ./node_modules/bootstrap/js/modal.js 1:0-17
 @ ./node_modules/bootstrap/dist/js/npm.js
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css

ERROR in ./node_modules/bootstrap/js/carousel.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/node_modules/bootstrap/js'
 @ ./node_modules/bootstrap/js/carousel.js 1:0-17
 @ ./node_modules/bootstrap/dist/js/npm.js
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css

ERROR in ./node_modules/bootstrap/js/button.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/node_modules/bootstrap/js'
 @ ./node_modules/bootstrap/js/button.js 1:0-17
 @ ./node_modules/bootstrap/dist/js/npm.js
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css

ERROR in ./node_modules/bootstrap/js/tooltip.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/node_modules/bootstrap/js'
 @ ./node_modules/bootstrap/js/tooltip.js 1:0-17
 @ ./node_modules/bootstrap/dist/js/npm.js
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css

ERROR in ./node_modules/bootstrap/js/dropdown.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/node_modules/bootstrap/js'
 @ ./node_modules/bootstrap/js/dropdown.js 1:0-17
 @ ./node_modules/bootstrap/dist/js/npm.js
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css

ERROR in ./node_modules/bootstrap/js/collapse.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/node_modules/bootstrap/js'
 @ ./node_modules/bootstrap/js/collapse.js 1:0-17
 @ ./node_modules/bootstrap/dist/js/npm.js
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css

ERROR in ./node_modules/bootstrap/js/popover.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/node_modules/bootstrap/js'
 @ ./node_modules/bootstrap/js/popover.js 1:0-17
 @ ./node_modules/bootstrap/dist/js/npm.js
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css

ERROR in ./node_modules/bootstrap/js/scrollspy.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/node_modules/bootstrap/js'
 @ ./node_modules/bootstrap/js/scrollspy.js 1:0-17
 @ ./node_modules/bootstrap/dist/js/npm.js
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css

ERROR in ./node_modules/bootstrap/js/tab.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/node_modules/bootstrap/js'
 @ ./node_modules/bootstrap/js/tab.js 1:0-17
 @ ./node_modules/bootstrap/dist/js/npm.js
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css

ERROR in ./node_modules/bootstrap/js/affix.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/node_modules/bootstrap/js'
 @ ./node_modules/bootstrap/js/affix.js 1:0-17
 @ ./node_modules/bootstrap/dist/js/npm.js
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css

ERROR in ./node_modules/bootstrap/js/transition.js
Module not found: Error: Can't resolve 'jquery' in '/usr/src/app/cassandra-reaper/src/ui/node_modules/bootstrap/js'
 @ ./node_modules/bootstrap/js/transition.js 1:0-17
 @ ./node_modules/bootstrap/dist/js/npm.js
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css

ERROR in multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css
Module not found: Error: Can't resolve 'metisMenu' in '/usr/src/app/cassandra-reaper/src/ui'
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css deps[9]

ERROR in multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css
Module not found: Error: Can't resolve 'metisMenu.css' in '/usr/src/app/cassandra-reaper/src/ui'
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css deps[4]

ERROR in multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css
Module not found: Error: Can't resolve 'rxjs' in '/usr/src/app/cassandra-reaper/src/ui'
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css deps[11]

ERROR in ./app/observable.js
Module not found: Error: Can't resolve 'rxjs' in '/usr/src/app/cassandra-reaper/src/ui/app'
 @ ./app/observable.js 8:12-27
 @ ./app/index.js
 @ multi ./app/index.js

ERROR in multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css
Module not found: Error: Can't resolve 'sb-admin-2' in '/usr/src/app/cassandra-reaper/src/ui'
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css deps[10]

ERROR in multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css
Module not found: Error: Can't resolve 'sb-admin-2.css' in '/usr/src/app/cassandra-reaper/src/ui'
 @ multi sb-admin-2.css bootstrap.css theme.css font-awesome.css metisMenu.css style.scss jquery react bootstrap metisMenu sb-admin-2 rxjs bootstrap-multiselect.css datatables.net-bs.css react-datepicker.css deps[0]
Child html-webpack-plugin for "events.html":
     1 asset
    Entrypoint undefined = events.html
    [0] ./node_modules/html-webpack-plugin/lib/loader.js!./app/html_template.ejs 2.39 KiB {0} [built]
    [2] (webpack)/buildin/global.js 472 bytes {0} [built]
    [3] (webpack)/buildin/module.js 497 bytes {0} [built]
        + 1 hidden module
Child html-webpack-plugin for "index.html":
     1 asset
    Entrypoint undefined = index.html
    [0] ./node_modules/html-webpack-plugin/lib/loader.js!./app/html_template.ejs 2.39 KiB {0} [built]
    [2] (webpack)/buildin/global.js 472 bytes {0} [built]
    [3] (webpack)/buildin/module.js 497 bytes {0} [built]
        + 1 hidden module
Child html-webpack-plugin for "login.html":
     1 asset
    Entrypoint undefined = login.html
    [0] ./node_modules/html-webpack-plugin/lib/loader.js!./app/html_template.ejs 2.39 KiB {0} [built]
    [2] (webpack)/buildin/global.js 472 bytes {0} [built]
    [3] (webpack)/buildin/module.js 497 bytes {0} [built]
        + 1 hidden module
Child html-webpack-plugin for "repair.html":
     1 asset
    Entrypoint undefined = repair.html
    [0] ./node_modules/html-webpack-plugin/lib/loader.js!./app/html_template.ejs 2.39 KiB {0} [built]
    [2] (webpack)/buildin/global.js 472 bytes {0} [built]
    [3] (webpack)/buildin/module.js 497 bytes {0} [built]
        + 1 hidden module
Child html-webpack-plugin for "schedules.html":
     1 asset
    Entrypoint undefined = schedules.html
    [0] ./node_modules/html-webpack-plugin/lib/loader.js!./app/html_template.ejs 2.39 KiB {0} [built]
    [2] (webpack)/buildin/global.js 472 bytes {0} [built]
    [3] (webpack)/buildin/module.js 497 bytes {0} [built]
        + 1 hidden module
Child html-webpack-plugin for "segments.html":
     1 asset
    Entrypoint undefined = segments.html
    [0] ./node_modules/html-webpack-plugin/lib/loader.js!./app/html_template.ejs 2.39 KiB {0} [built]
    [2] (webpack)/buildin/global.js 472 bytes {0} [built]
    [3] (webpack)/buildin/module.js 497 bytes {0} [built]
        + 1 hidden module
Child html-webpack-plugin for "snapshot.html":
     1 asset
    Entrypoint undefined = snapshot.html
    [0] ./node_modules/html-webpack-plugin/lib/loader.js!./app/html_template.ejs 2.39 KiB {0} [built]
    [2] (webpack)/buildin/global.js 472 bytes {0} [built]
    [3] (webpack)/buildin/module.js 497 bytes {0} [built]
        + 1 hidden module
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! cassandra-reaper-ui@0.2.0 minimize: `webpack -p --progress --colors --devtool none`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the cassandra-reaper-ui@0.2.0 minimize script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-03-26T19_20_05_949Z-debug.log
[ERROR] Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 2 (Exit value: 2)
    at org.apache.commons.exec.DefaultExecutor.executeInternal (DefaultExecutor.java:404)
    at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:166)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:804)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:751)
    at org.codehaus.mojo.exec.ExecMojo.execute (ExecMojo.java:313)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Reaper for Apache Cassandra project 3.2.0-SNAPSHOT:
[INFO] 
[INFO] Reaper for Apache Cassandra project ................ SUCCESS [  0.317 s]
[INFO] Reaper for Apache Cassandra server ................. FAILURE [ 27.086 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  27.470 s
[INFO] Finished at: 2022-03-26T19:20:06Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.6.0:exec (exec-npm-run) on project cassandra-reaper: Command execution failed. Process exited with an error: 2 (Exit value: 2) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :cassandra-reaper
make: *** [Makefile:17: package] Error 1
ERROR: 2
```

Since I build with docker I assume it's environment independent and you must have the same problem?

**Comment by ossarga** [comment]
@adejanovski, I have a [branch](https://github.com/thelastpickle/cassandra-reaper/compare/master...anthony/ISSUE-890/build_jar_when_creating_packages) with changes on it that incorporates @simonfontana's original fix and addresses the compile issue that was encountered.

Should we close this pull request and I'll open a new one with all the changes?

**Comment by adejanovski** [comment]
> Should we close this pull request and I'll open a new one with all the changes?

@ossarga, yes, works for me.

**Pull request closed without merging**